### ISSUE TITLE
library panics when trying to marshal a nil *time.Time struct member

### ISSUE THREAD

**Issue by twbryan** [guregu/dynamo#99]
The default AWS dynamodb library marshals a nil pointer as "null" (the desired behavior, as would omitting the field). This library panics:

```
panic: value method time.Time.MarshalText called using nil *Time pointer

goroutine 1 [running]:
time.(*Time).MarshalText(0x0, 0x49b7300, 0x0, 0x5386740, 0x0, 0x1)
	<autogenerated>:1 +0x93
github.com/guregu/dynamo.marshal(0x49b7300, 0x0, 0x0, 0x0, 0x49b7300, 0x0, 0x0)
	/Users/bryan/go/pkg/mod/github.com/guregu/dynamo@v1.2.1/encode.go:117 +0x23e
github.com/guregu/dynamo.marshalStruct(0x499f680, 0xc0004da2d0, 0x99, 0x4b60760, 0x499f680, 0xc0004da2d0)
	/Users/bryan/go/pkg/mod/github.com/guregu/dynamo@v1.2.1/encode.go:91 +0x4ef
github.com/guregu/dynamo.marshalItem(0x499f680, 0xc0004da2d0, 0x199, 0x499f601, 0x499f680)
	/Users/bryan/go/pkg/mod/github.com/guregu/dynamo@v1.2.1/encode.go:33 +0x28a
github.com/guregu/dynamo.marshalItem(0x486d500, 0xc0004da1e0, 0x0, 0x0, 0xc00012a400)
	/Users/bryan/go/pkg/mod/github.com/guregu/dynamo@v1.2.1/encode.go:31 +0x323
github.com/guregu/dynamo.Table.Put(...)
	/Users/bryan/go/pkg/mod/github.com/guregu/dynamo@v1.2.1/put.go:24
```

**Comment by tibraga** [comment]
I get exact error too...

**Comment by guregu** [comment]
Thank you for the report. I've found a workaround and it should be fixed soon.

**Comment by guregu** [comment]
I've fixed this in version `v1.3.0`. Now these types will be automatically omitted if they are nil, similar to empty strings.

**Comment by twbryan** [comment]
can confirm the fix works :) thanks
